batch2/batch/worker.py

```diff
@@ -212,6 +212,10 @@ def step(self, name, **kwargs):
 
             async with self.step('runtime', state=None):
                 async with worker.cpu_sem(self.cpu_in_mcpu):
+                    if self.name == 'main':
+                        self.job.start_time = time.time()
```

Comment:
Using self.timing['runtime']['start_time'] will be more consistent.

Reply:
My concern was with the cpu_semaphore acquisition not matching the runtime start time. If I switch the order of the cpu_semaphore acquisition with the runtime start, then I'm okay doing this. Otherwise, I worry that for example an 8 core job that never acquires the semaphore, this will be wrong.

Reply:
Yeah, do that, that was a mistake on my end, I didn't mean the runtime to include the wait for resources.